src/Controller/SantaController.php

```diff
@@ -128,18 +141,29 @@ public function message(Rudolph $rudolph, Request $request, string $application)
         $message = $session->get('message');
         $notes = $session->get('notes');
 
+        $form = $this->createForm(MessageType::class, null, [
+            'message' => $message,
+            'selected-users' => $selectedUsers,
+        ]);
+
         $errors = [];
+        $form->handleRequest($request);
 
         if ($request->isMethod('POST')) {
-            $message = trim($request->request->get('message'));
-            $notes = array_map('trim', $request->request->all('notes'));
-
-            if ($messageError = $this->validateMessage($message)) {
-                $errors['message'] = $messageError;
+            if ($form->isSubmitted() && $form->isValid()) {
+                $message = $form->getData()['message'] ?? '';
+                foreach ($form->getData() as $data => $note) {
+                    if (str_contains($data, 'notes-')) {
+                        $notes[] = $note;
+                    }
+                }
```

Comment:
code inside the if block `if (!$errors) {` should be moved here because it's better to let the symfony validate the data and ensure there is no error